### PR TITLE
Fix security issue with CORS proxy

### PR DESCRIPTION
**Pull request by HoldYourWaffle**
### TL;DR: Error handling is broken when loading an image, how do I forward an error at the `TODO`-comments I placed to the `catch`-callback of the `TesseractJob`?

During my adventures while using Tesseract.js I often had the problem of Tesseract.js failing silently when trying to recognize an image. After digging through the source code I found out that the error handling in this area is so to speak 'incomplete'.
There are 2 `onerror` callbacks missing, and the XHR one for some reason tries to load an image through an external proxy (more on that later). There's also no checking for HTTP status codes, which causes a completely silent fail when something like a 404 happens (this was my problem).

I tried to fix it by having it call the `catch`-callback on the `TesseractJob`, but I couldn't find out how to do that, so for now I have just added `TODO`-comments where the error-handling should be fixed. I'm not creating this PR with the intent of it being merged in it's current state, I'll be glad to fix the error handling myself with the right pointers.

I have also removed the [crossorigin.me](http://crossorigin.me/) proxy. The main reason being security, there's a reason for CORS being prohibited by default. We're also trusting a random proxy with potentially sensitive data, which isn't really a good idea. There are of course valid reasons for using such a proxy, but that decision should be made by the user and not some library they are using.
[crossorigin.me](http://crossorigin.me/) also explicitly states that the proxy is not intended for production use, and as far as I can tell there's no distinction between debug and production in that piece of code.

### CODE REVIEW

**Comment by HoldYourWaffle** [comment]
Thanks for taking a look at this!

> In fact, right now I am working on tessearct.js v2 (you can check develop branch), and it will be better if we add code there instead of master branch.

Makes sense.

> But if you still prefer error handling in the v1, please let me know and I will take a deep look.

Depends on how long it'll be until v2 is released, but I think we should definitely remove the proxy in v1 (I can create a separate PR for this if you want).

**Comment by jeromewu** [comment]
@HoldYourWaffle 

I have created a pre-release version of tesseract.js, you can install with following command:

```shell
$ npm install --save tesseract.js@next
```

You can check if it fits your requirements :)

**Comment by HoldYourWaffle** [comment]
Seems like the security problem was fixed, great news! I still see some missing error (`catch`) handlers, but I'll have to take a closer look to determine exactly what's missing where. I'll probably create another PR for this some time in the future.

I'd still highly recommend merging this into the v1 branch, as users won't be updated automatically to the v2 version without the security fix. I have changed the base branch of this PR accordingly.
